src/lib.rs

```diff
@@ -996,6 +996,49 @@ impl Build {
             }
         }
 
+        if self.cuda && cfg!(feature = "cudart") {
+            match which(&self.get_compiler().path) {
+                None => (),
+                Some(nvcc) => {
+                    // Try to figure out the -L search path. If it fails,
+                    // it's on user to specify one by passing it through
+                    // RUSTFLAGS environment variable.
+                    let mut libtst = false;
+                    let mut libdir = nvcc.canonicalize().unwrap();
```

Comment:
I've found historically that `canonicalize()` can fail for weird reasons, but is that necesary to do here? We already find `nvcc` in `PATH` so I don't think we may need to canonicalize here?

Reply:
Note that `PATH` is evaluated only in the case `nvcc` here is a "single-word" command. Otherwise one can set the NVCC environment variable to explicit path to whatever you want to call `nvcc`. The idea behind `canonicalize()` is to handle the case when the environment variable is set to **relative** path. One can argue that it's an edge case that is not worth covering. If you want to say "scrap it," then just say the word:-)

Reply:
If `nvcc` is set to a relative path is there an issue with passing a relative path as a `-L` argument? I would naively expect that to work ok, which I think means that the `canonicalize()` probably isn't necessary here.

Reply:
It's surely just me, but I tend to think of unlike cases. Let's say you want to compile with NVCC set to './nvcc'. Of course nobody in their right mind would actually do it, but it's not really an excuse to disregard the possibility. Without `canonicalize()` the current code will break. Alternative can be to not rely on `libdir.pop(), libdir.pop()`, but do `libdir.pop(), libdir.push("..")`... This way most unlike case is covered without `canonicalize()`. Let me test how it works...

Reply:
`canonicalize()` is gone.